### ISSUE TITLE
Generate a list of all templates for resources

### ISSUE THREAD

**Issue by tbe1** [liimaorg/liima#231]
Currently it is only possible to access already generated templates via the ${templates} hash. Templates which are not generated at the time are not accessible. 
It should be possible to get a list of of all templates (Name, Path without Content) for all resources and all resources relations if possible grouped by resource type.

**Comment by yvespp** [comment]
Result of my test template:
```
Templates:
<#list templates?keys as key> 
${key} = ${templates[key].name}, ${templates[key].path}, ${templates[key].content}
</#list>

Templates of a relation jboss7Management_Ldap:
<#list jboss7Management_Ldap.templates?keys as key> 
${key} = ${templates[key].name}, ${templates[key].path}, ${templates[key].content}, ${templates[key].isRelationTemplate}
</#list>
```

Rendered template:
```
Templates:
run_app.conf = run_app.conf, configuration/run_app.conf, 
bean-instance-pools_adam = bean-instance-pools_adam, bean-instance-pools_adam.tmp, 
application-users.properties = application-users.properties, configuration/props/application-users.properties, 

Templates of a relation jboss7Management_Ldap:
templateOnRel= asdf, asdf, , 
templateOnRes = , , , 
```
Issues:
* template.content is always empty.
* Templates defined on the resource type relation are not shown.
* Templates defined on the related resource have no values (`templateOnRes = , , ,` in the rendered template). Didn't tests templates on the related resource type.
* `isRelationTemplate` is always empty.

**Comment by RetGal** [comment]
Fixed in #378

**Comment by yvespp** [comment]
If a template is defined on relateded resource (not on the relation), then it doesn't show up in the list of relations templates. The template seems to be in the list but all values are empty. E.g.:
```
Templates of a relation jboss7Management_Ldap:
<#list jboss7Management_Ldap.templates?keys as key> 
${key} = name: ${templates[key].name}, name: ${templates[key].path}, name: ${templates[key].content}, ${templates[key].isRelationTemplate}
</#list>
```
jboss7Management_Ldap has a template directly defined on it. Test generation gives the following error:
```
The following has evaluated to null or missing: ==> templates[key] [in template "test" at line 3, column 18]
---- Tip: It's the final [] step that caused this error, not those before it.
---- Tip: If the failing expression is known to be legally refer to something that's sometimes null or missing, either specify a default value like myOptionalVar!myDefault, or use <#if myOptionalVar??>when-present<#else>when-missing</#if>. (These only cover the last step of the expression; to cover the whole expression, use parenthesis: (myOptionalVar.foo)!myDefault, (myOptionalVar.foo)?? ----
---- FTL stack trace ("~" means nesting-related): - Failed at: ${templates[key].name} [in template "test" at line 3, column 16] ----
```
